hnn_core/tests/test_mpi_child.py

```diff
@@ -9,8 +12,70 @@
 from hnn_core.parallel_backends import MPIBackend
 
 
+def test_read_stderr():
+    """Test the _read_stderr handler for processing data and signals"""
+    (pipe_stderr_r, pipe_stderr_w) = os.pipe()
+    stderr = os.fdopen(pipe_stderr_w, 'w')
+    backend = MPIBackend()
+
+    stderr.write("test_data")
+    stderr.flush()
+    backend._read_stderr(pipe_stderr_r, selectors.EVENT_READ)
+    assert backend.proc_data_bytes == "test_data".encode()
+    backend.proc_data_bytes = b''
+
+    stderr.write("@")
+    stderr.flush()
+    with pytest.raises(ValueError, match="Invalid signal start"):
+        backend._read_stderr(pipe_stderr_r, selectors.EVENT_READ)
+
+    stderr.write("@end_of_data:")
+    stderr.flush()
+    with pytest.raises(ValueError, match="Invalid signal start"):
+        backend._read_stderr(pipe_stderr_r, selectors.EVENT_READ)
+
+    stderr.write("@end_of_data:@")
+    stderr.flush()
+    with pytest.raises(ValueError, match="Completion signal from child MPI "
+                       "process did not contain data length."):
+        backend._read_stderr(pipe_stderr_r, selectors.EVENT_READ)
+
+    stderr.write("blahblah@end_of_data:1000@blah")
```

Comment:
The `blah` at the end would not be anticipated irl as mentioned in an earlier comment. Though, it helps to test the separation of the signal from the data in `_read_stderr()`

Reply:
maybe we're missing the test case of `blahblah@end_of_data:100` ? This should fail, right?

Reply:
That seems like a good test to add.